shaders/src/common_types.fs

```diff
@@ -22,3 +22,10 @@
 
 #define float3x3 mat3
 #define float4x4 mat4
+
+struct ShadowData {
+    highp mat4 spotlightFromWorldMatrix;
```

Comment:
This won't work on Adreno, `highp` is ignored inside `struct`...

Reply:
Although I guess that will work inside a UBO? If so you should probable leave a comment explaining this struct MUST be used in a UBO.